ossdbtoolsservice/language/completion/mysqlcompleter.py

```diff
@@ -0,0 +1,991 @@
+        if not template:
+            return '()'
+        elif usage == 'call' and len(args) < 2:
+            return '()'
+        elif usage == 'call' and func.has_variadic():
+            return '()'
+        multiline = usage == 'call' and len(args) > self.call_arg_oneliner_max
+        max_arg_len = max(len(a.name) for a in args) if multiline else 0
+        args = (
+            self._format_arg(template, arg, arg_num + 1, max_arg_len)
+            for arg_num, arg in enumerate(args)
+        )
+        if multiline:
+            return '(' + ','.join('\n    ' + a for a in args if a) + '\n)'
+        else:
+            return '(' + ', '.join(a for a in args if a) + ')'
+
+    def _format_arg(self, template, arg, arg_num, max_arg_len):
+        if not template:
+            return None
+        if arg.has_default:
+            arg_default = 'NULL' if arg.default is None else arg.default
+            # Remove trailing ::(schema.)type
+            arg_default = arg_default_type_strip_regex.sub('', arg_default)
+        else:
+            arg_default = ''
+        return template.format(
+            max_arg_len=max_arg_len,
+            arg_name=self.case(arg.name),
+            arg_num=arg_num,
+            arg_type=arg.datatype,
+            arg_default=arg_default
+        )
+
+    def _make_cand(self, tbl, do_alias, suggestion, arg_mode=None):
+        """Returns a Candidate namedtuple.
+
+        :param tbl is a SchemaObject
+        :param arg_mode determines what type of arg list to suffix for functions.
+        Possible values: call, signature
+
+        """
+        cased_tbl = self.case(tbl.name)
+        if do_alias:
+            alias = self.alias(cased_tbl, suggestion.table_refs)
+        synonyms = (cased_tbl, generate_alias(cased_tbl))
+        maybe_alias = (' ' + alias) if do_alias else ''
+        maybe_schema = (self.case(tbl.schema) + '.') if tbl.schema else ''
+        suffix = self._arg_list_cache[arg_mode][tbl.meta] if arg_mode else ''
+        if arg_mode == 'call':
+            display_suffix = self._arg_list_cache['call_display'][tbl.meta]
+        elif arg_mode == 'signature':
+            display_suffix = self._arg_list_cache['signature'][tbl.meta]
+        else:
+            display_suffix = ''
+        item = maybe_schema + cased_tbl + suffix + maybe_alias
+        display = maybe_schema + cased_tbl + display_suffix + maybe_alias
+        prio2 = 0 if tbl.schema else 1
+        return Candidate(item, synonyms=synonyms, prio2=prio2, display=display, schema=tbl.schema_name)
+
+    def get_table_matches(self, suggestion, word_before_cursor, alias=False):
+        tables = self.populate_schema_objects(suggestion.schema, 'tables')
+        tables.extend(SchemaObject(tbl.name) for tbl in suggestion.local_tables)
+
+        # Unless we're sure the user really wants them, don't suggest the
+        # pg_catalog tables that are implicitly on the search path
+        if not suggestion.schema and (
```

Comment:
This is also pg specific code, we should not add it if we are not using it as part of our literal completion. This is a huge file, we can miss stuff that is not needed if we just copy the whole file